### ISSUE TITLE
Leaks?

### ISSUE THREAD

**Issue by Fabezi** [malcommac/Hydra#40]
Played around with this library a bit and noticed Xcode is reporting quite a few memory leaks using the latest version and with a simple function:

```
func testPromise() -> Promise<Int> {
        return Promise<Int>(in: .utility, { (resolve, reject, _) in
            resolve(5)
        })
}

testPromise().then { result in
        print(result)
}
```

![screen shot 2017-07-30 at 17 03 32](https://user-images.githubusercontent.com/3667836/28754262-cfa6d38a-754a-11e7-8d36-8c6852950c6b.png)

**Comment by terhechte** [comment]
I've fixed the leak in this PR: https://github.com/malcommac/Hydra/pull/48

**Comment by malcommac** [comment]
Fixed in #48 (https://github.com/malcommac/Hydra/pull/48/commits/2d8c0ce2d88ca49f7e4286661b92359996024417)
